examples/improved_wgan.py

```diff
@@ -71,10 +71,23 @@ def gradient_penalty_loss(y_true, y_pred, averaged_samples, gradient_penalty_wei
     Note that this loss function requires the original averaged samples as input, but Keras only supports passing
     y_true and y_pred to loss functions. To get around this, we make a partial() of the function with the
     averaged_samples argument, and use that for model training."""
-    gradients = K.gradients(K.sum(y_pred), averaged_samples)
-    gradient_l2_norm = K.sqrt(K.sum(K.square(gradients)))
+    # first get the gradients:
+    #   assuming: - that y_pred has dimensions (batch_size, 1)
+    #             - averaged_samples has dimensions (batch_size, nbr_features)
+    # gradients afterwards has dimension (batch_size, nbr_features), basically
+    # a list of nbr_features-dimensional gradient vectors
+    gradients = K.gradients(y_pred, averaged_samples)
+    # compute the euclidean norm by squaring ...
+    gradients_sqr = K.square(gradients)[0]

```

Comment:
I think we should select the first element of the gradient list before taking the square root:
```python
gradients = K.gradients(y_pred, averaged_samples)[0]
gradients_sqr = K.square(gradients)
```
Maybe a *tiny* performance improvement 🚀

Reply:
Oh that's actually the square function adding this dimension for some reason. Pretty weird but your suggestion would actually change the behaviour.

Reply:
> Oh that's actually the square function adding this dimension for some reason.

`K.square` should compute the square element-wise so I'm pretty sure it doesn't change the shape: https://keras.io/backend/#square
Or am I missing something?

Reply:
Ah you were right ;) I encountered this problem already in my own implementation and thought it was because of some strange implementation of the square function, but didn't gave it too much thought. Didn't noticed that actually the gradients function returns a list of tensors, this list being the additional dimension.